scan.go

```diff
@@ -248,6 +248,7 @@ type ZoneParser struct {
 	includeDepth uint8
 
 	includeAllowed bool
+	generateAllowed bool
```

Comment:
I’d prefer we flipped this to `generateDisallowed`, that way we don’t have to initialise it in `NewZoneParser`. It’s not a big deal though and this could be more consistent.